docs/clients/rust/queryable.rst

```diff
@@ -0,0 +1,148 @@
+.. _ref_rust_queryable:
+
+Using the ``Queryable`` macro
+-----------------------------
+
+The easiest way to unpack an EdgeDB query result is the built-in
+``Queryable`` macro from the ``edgedb-derive`` crate. This turns queries
+directly into Rust types without having to match on a ``Value`` (more in
+the section on the ``Value`` enum), cast to JSON, etc.
+
+.. code-block:: rust
+
+  #[derive(Debug, Deserialize, Queryable)]
+  pub struct QueryableAccount {
+      pub username: String,
+      pub id: Uuid,
+  }
+
+  let query = "select account {
+      username,
+      id
+      };";
+  let as_queryable_account: QueryableAccount = client
+      .query_required_single(query, &())
+      .await?;
+
+Note: Field order within the shape of the query matters when using the
+``Queryable`` macro. In the example below, a query is done in the order
+``id, username`` instead of ``username, id`` as defined in the struct:
```

Comment:
```suggestion
.. note::

    Field order within the shape of the query matters when using the
    ``Queryable`` macro. In the example below, we run a query with the order
    ``id, username`` instead of ``username, id`` as defined in the struct:
```